src/ReactiveUI.AndroidSupport/ControlFetcherMixin.cs

```diff
@@ -24,122 +21,182 @@ namespace ReactiveUI.AndroidSupport
-        public static T GetControl<T>(this View @this, [CallerMemberName]string propertyName = null)
-            where T : View => (T)GetCachedControl(propertyName, @this, () => @this.FindViewById(controlIds[propertyName.ToLowerInvariant()]).JavaCast<T>());
+        /// <returns>The return view.</returns>
+        public static T GetControl<T>(this Fragment fragment, [CallerMemberName] string propertyName = null)
+            where T : View => GetControl<T>(fragment.View, propertyName);
+
+        /// <summary>
+        /// Wires a control to a property.
+        /// </summary>
+        /// <param name="layoutHost">The layout view host.</param>
+        /// <param name="resolveMembers">The resolve members.</param>
+        public static void WireUpControls(this ILayoutViewHost layoutHost, ReactiveUI.ControlFetcherMixin.ResolveStrategy resolveMembers = ReactiveUI.ControlFetcherMixin.ResolveStrategy.Implicit)
+        {
+            var members = layoutHost.GetWireUpMembers(resolveMembers).ToList();
+            members.ForEach(m =>
+            {
+                try
+                {
+                    var view = layoutHost.View.GetControlInternal(m.GetResourceName());
+                    m.SetValue(layoutHost, view);
+                }
+                catch (Exception ex)
+                {
+                    throw new
+                        MissingFieldException("Failed to wire up the Property " + m.Name + " to a View in your layout with a corresponding identifier", ex);
+                }
+            });
+        }
 
         /// <summary>
-        /// A helper method to automatically resolve properties in an <see cref="Android.Support.V4.App.Fragment"/> to their respective elements in the layout.
-        /// This should be called in the Fragement's OnCreateView, with the newly inflated layout.
+        /// Wires a control to a property.
         /// </summary>
-        /// <param name="this">The fragment.</param>
-        /// <param name="inflatedView">The newly inflated <see cref="View"/> returned from Inflate.</param>
-        /// <param name="resolveMembers">The strategy used to resolve properties that either subclass <see cref="View"/>, have a <see cref="WireUpResourceAttribute"/> or have a <see cref="IgnoreResourceAttribute"/>.</param>
-        public static void WireUpControls(this Android.Support.V4.App.Fragment @this, View inflatedView, ResolveStrategy resolveMembers = ResolveStrategy.Implicit)
+        /// <param name="view">The view.</param>
+        /// <param name="resolveMembers">The resolve members.</param>
+        public static void WireUpControls(this View view, ReactiveUI.ControlFetcherMixin.ResolveStrategy resolveMembers = ReactiveUI.ControlFetcherMixin.ResolveStrategy.Implicit)
         {
-            var members = @this.GetWireUpMembers(resolveMembers);
+            var members = view.GetWireUpMembers(resolveMembers);
+
+            members.ToList().ForEach(m =>
```

Comment:
Given you have to do ToList() I think you should just do a normal regular `foreach (var member in members)` -- avoids the extra allocations.